lib/hd/keyorigin.js

```diff
@@ -0,0 +1,173 @@
+/*!
+ * keyorigin.js - hd key path object for bcoin
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('assert');
+const common = require('./common');
+const bio = require('bufio');
+const {inspectSymbol} = require('../utils');
+/**
+ * Helper class to represent hd key path for arbitrary wallets.
+ * @property {Number} fingerPrint - master key fingerprint (uint32)
+ * @property {Array} path - bip32 derivation path in uint32 array
+ */
+class KeyOriginInfo {
+  constructor(options) {
+    this.fingerPrint = -1;
+    this.path = [];
+    if (options) {
+      this.fromOptions(options);
+    }
+  }
+
+  fromOptions(options) {
+    assert(options, 'requires options');
+
+    if (options.fingerPrint !== undefined) {
```

Comment:
I think we typically test optional options with `... =! null` in `fromOptions()` functions around the codebase